mlir/lib/Conversion/GPUCommon/OpToFuncCallLowering.h

```diff
@@ -90,9 +91,13 @@ struct OpToFuncCallLowering : public ConvertOpToLLVMPattern<SourceOp> {
     return LLVM::LLVMFunctionType::get(resultType, operandTypes);
   }
 
-  StringRef getFunctionName(Type type) const {
-    if (isa<Float32Type>(type))
-      return f32Func;
+  StringRef getFunctionName(Type type, arith::FastMathFlags flag) const {
+    if (isa<Float32Type>(type)) {
+      if (arith::FastMathFlags::fast == flag && !f32FastFunc.empty())
```

Comment:
```suggestion
      if ((arith::FastMathFlags::afn & flag) && !f32FastFunc.empty())
```

We only need afn I believe (please add such test)

Reply:
Thanks, modified, the code diff please see : [link](https://github.com/llvm/llvm-project/pull/99890/files#diff-f9d186a175b0bff9446d83449269fefcf90dd8f9171227afa2ee4fbbd7ce2ef6R105)
the related test case please see : [link](https://github.com/llvm/llvm-project/pull/99890/files#diff-8b41948f461ca1b7b892e0ab0488723021ea6b47df5987c6f442669b6b0652d6R525)

Reply:
I still use `arith::FastMathFlags::afn == flag` instead of `arith::FastMathFlags::afn & flag`. The reason is that the binary of `afn` is `1000000` (decimal is 64) , if flag is `fast` whose binary is `1111111` (decimal is 127) , the result of `&` is still true.

Reply:
Perhaps i get your point, `fast` should also imply `afn`, so `&` is necessary. Modified to `&`, please see : [link](https://github.com/llvm/llvm-project/pull/99890/files#diff-f9d186a175b0bff9446d83449269fefcf90dd8f9171227afa2ee4fbbd7ce2ef6R105)

Related test case please see : [link](https://github.com/llvm/llvm-project/pull/99890/files#diff-8b41948f461ca1b7b892e0ab0488723021ea6b47df5987c6f442669b6b0652d6R264) and [link](https://github.com/llvm/llvm-project/pull/99890/files#diff-8b41948f461ca1b7b892e0ab0488723021ea6b47df5987c6f442669b6b0652d6R910)

please take a look again :)